packages/purpose-process/src/services/purposeService.ts

```diff
@@ -1591,23 +1615,28 @@ const performUpdatePurpose = async (
     mode,
     updateContent,
   }:
-    | { mode: "Deliver"; updateContent: purposeApi.PurposeUpdateContent }
+    | {
+        mode: "Deliver";
+        updateContent:
+          | purposeApi.PurposeUpdateContent
+          | purposeApi.PATCH_PurposeUpdateContent;
+      }
     | {
         mode: "Receive";
         updateContent: purposeApi.ReversePurposeUpdateContent;
       },
   authData: UIAuthData,
   readModelService: ReadModelService,
   correlationId: CorrelationId,
-  repository: {
-    createEvent: (createEvent: CreateEvent<PurposeEvent>) => Promise<unknown>;
-  }
+  repository: ReturnType<typeof eventRepository<PurposeEvent>>
+): Promise<
+  WithMetadata<{ purpose: Purpose; isRiskAnalysisValid: boolean }>
   // eslint-disable-next-line max-params
-): Promise<{ purpose: Purpose; isRiskAnalysisValid: boolean }> => {
+> => {
   const purpose = await retrievePurpose(purposeId, readModelService);
   assertPurposeIsDraft(purpose.data);
 
```

Comment:
Here I would do something as follows, to make sure we don't forget to update some optional prop - then, below, never use `updateContent.` but always use the destructured values from here:

```suggestion
  const {
    title,
    description,
    isFreeOfCharge,
    freeOfChargeReason,
    dailyCalls,
    riskAnalysisForm,
    ...rest
  } = updateContent;

  void (rest satisfies Record<string, never>); // To make sure we extract all the updated fields, even optional ones
```

Reply:
`riskAnalysisForm` is a property that is missing in an type union branch, it cannot be destructured directly.
One solution is to not destructure it and modify te type assertion this way 

```typescript
  void (rest satisfies
    | Record<string, never>
    | {
        riskAnalysisForm?: purposeApi.PurposeUpdateContent["riskAnalysisForm"];
      }); // To make sure we extract all the updated fields, even optional ones
```

Not beautiful, but it should work. WDYT?

Reply:
Ok! Then we could use this approach for all patch calls, so that we make sure to not forget some props